pkg/storaged/block/format-dialog.jsx

```diff
@@ -517,13 +538,13 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
 
                 function format() {
                     if (create_partition) {
-                        if (vals.type == "dos-extended")
+                        if (type == "dos-extended")
                             return block_ptable.CreatePartition(start, vals.size, "0x05", "", { });
-                        else if (vals.type == "empty")
-                            return block_ptable.CreatePartition(start, vals.size, "", "", { });
+                        else if (type == "empty")
+                            return block_ptable.CreatePartition(start, vals.size, partition_type, "", { });
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19826-20240112-132957-b00e3528-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)